content/docs/hoon/reference/rune/bar.md

```diff
@@ -553,156 +762,222 @@ with `$()`.
 `|*` can be used to make wet gates that produce structures, but this usage is
 discouraged in favor of `|$`.
 
-##### Examples
+#### Examples
 
 Wet and dry gates in a nutshell:
 
 ```
-~zod:dojo> =foo |=([a=* b=*] [b a])
+> =foo |=([a=* b=*] [b a])
 
-~zod:dojo> =bar |*([a=* b=*] [b a])
+> =bar |*([a=* b=*] [b a])
 
-~zod:dojo> (foo %cat %dog)
+> (foo %cat %dog)
 [6.778.724 7.627.107]
 
-~zod:dojo> (bar %cat %dog)
+> (bar %cat %dog)
 [%dog %cat]
 ```
 
 The dry gate does not preserve the type of `a` and `b`; the wet
 gate does.
 
-### `|=` "bartis"
+---
 
-Produce a gate (a one-armed core with a sample).
+## `|=` "bartis"
 
-##### Syntax
+Produce a gate (a one-armed core with a sample).
 
-Regular:
+#### Syntax
+
+Two arguments, fixed.
+
+<table>
+<tr><th>Form</th><th>Syntax</th></tr>
+<tr>
+<td>Tall</td>
+<td>
+<pre>
+|=  a
+b
+</pre>
+</td>
+</tr>
+<tr>
+<td>Wide</td>
+<td>
+<pre>
+|=(a b)
+</pre>
+</td>
+</tr>
+<tr>
+<td>Irregular</td>
+<td>None.</td>
+</tr>
+</table>
+
+#### AST
 
 ```hoon
-|=  a=spec  b=hoon
-```
-
-AST:
-
-```
 [%brts p=spec q=hoon]
 ```
 
-##### Definition
-
-The product of a `|=` expression is a dry **gate**, i.e., a Hoon function.
-
-`p` defines the gate sample type -- i.e., the input value type -- and usually includes a sample name assignment (e.g., `a=@`).  `q` is an expression that determines the output value of the gate.
-
-##### Expands to
+#### Expands to
 
 ```hoon
 =|  a=spec
 |%  ++  $  b=hoon
 --
```

Comment:
The `^~` is actually here, so it'd be `=+  ^~(*a)  |% ...`. Not sure if you want to go to that level of detail ...

Reply:
is there an easy way to figure out these expansions? I've just left these as they were cos they were so painful to validate. Not sure these are even useful to people tbh

Reply:
I would prefer to strip them cos for 99% of people they'll be more confusing than illuminating but maybe it's desirable to have them

Reply:
I tries `~(open ap (ream '(add 1 1)'))`, is that how you'd do it? Is there a way to get something more readable than straight AST?